ZEP000-wf-stream-processing.md

```diff
@@ -0,0 +1,1421 @@
+
+On processing a command, the processor may produces a bunch of events until reaching a wait-state. Since all events must be written at once (i.e. atomically), the batch can exceed the max. message size.
+
+Another problem may be infinite loops in the workflow without reaching a wait state. 
+
+# Prior art
+[prior-art]: #prior-art
+
+<!--
+
+Discuss prior art, both the good and the bad, in relation to this proposal. A few examples of what this can include are:
+
+- For language, library, tools, and UI proposals: Does this feature exist in other tools/products and what experience have their community had?
+- For community proposals: Is this done by some other community and what were their experiences with it?
+- For other teams: What lessons can we learn from what other communities have done here?
+- Papers: Are there any published papers or great posts that discuss this? If you have some relevant papers to refer to, this can serve as a more detailed theoretical background.
+
+This section is intended to encourage you as an author to think about the lessons from other languages, provide readers of your ZEP with a fuller picture. If there is no prior art, that is fine - your ideas are interesting to us whether they are brand new or if it is an adaptation from other languages.
+
+Note that while precedent set by other products is some motivation, it does not on its own motivate a ZEP.
+
+-->
+
+Event Sourcing in general:
+* https://martinfowler.com/eaaDev/EventSourcing.html
+* https://martinfowler.com/eaaDev/EventNarrative.html
+
+Apache Kafka and Stream Processing:
+* https://www.confluent.io/blog/making-sense-of-stream-processing/
+* https://www.confluent.io/blog/event-sourcing-cqrs-stream-processing-apache-kafka-whats-connection/
+
+# Out of scope
+[out-of-scope]: #out-of-scope
+
+<!--
+
+Call out anything which is explicitly not part of this ZEP.
+
+-->
+
+This document does not describe, and assumes you already know:
+* stream processing in general and event sourcing, see [prior art](#prior-art)
+* the BPMN spec
+
+This proposal does not deal with:
+* other failure cases (of updating between different zeebe versions) than
+   business logic changes during stream processing
+* comprehesively documenting all business logic of the workflow processing of
+   zeebe (e.g. detailing all required state)
+* adjustments to the concept of blacklisting workflow instances   
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+<!--
+
+- What parts of the design do you expect to resolve through the ZEP process before this gets merged?
+- What parts of the design do you expect to resolve through the implementation of this feature before stabilization?
+- What related issues do you consider out of scope for this ZEP that could be addressed in the future independently of the solution that comes out of this ZEP?
+
+-->
+
+The main questions at this time:
+- Does this proposal really resolve the updatability problem of the engine? =>
```

Comment:
Is this still up to date?

Reply:
No it has been answered in the guide level explanation with an example. I'll remove this line